tests/lib/nsxv_test.py

```diff
@@ -861,15 +956,16 @@ def testNsxvStrForDestinationIpv4Mismatch(self):
     # parse the xml
     root = ET.fromstring(sections[1])
 
-    # check destination has only Ipv6
-    destination_address = root.findall('./rule/destinations/destination')
-    self.assertIsNotNone(destination_address)
-    for dest_addr in destination_address:
-      addr_type = dest_addr.find('type').text
-      self.assertEquals(addr_type, 'Ipv6Address')
+    source_addr = root.findall('./rule/sources/source')
+    dest_addr = root.findall('./rule/destinations/destination')
 
-    self.naming.GetNetAddr.assert_has_calls(
-        [mock.call('SOME_HOST'), mock.call('GOOGLE_DNS')])
+    return source_addr, dest_addr
+
+  def verifyAddressType(self, address, type):
```

Comment:
Let me try to expand on this a bit.

in an earlier section (which I commented on (testMixedToMixed and testMixedToAny)), this function was duplicated in line, because the assertEquals was replaced with "self.assertIn(saddr_type, ('Ipv4Address', 'Ipv6Address'))".

my comment originally was, if this function is changed to make type a list instead of a string, the function could be updated to be assertIn, that way that function can be used for all the tests.

However in my comment that I made today earlier, that assertIn doesn't test exactly what we need (because it doesn't ensure both are true), if that can be solved and the refactor still works I think it should be done.

Hopefully that helps explain my intention a bit better.